generator/handlebars/kss_handlebars_generator.js

```diff
@@ -270,15 +272,17 @@ kssHandlebarsGenerator.generatePage = function(styleguide, sections, root, secti
       console.log(' - homepage');
     }
     // Ensure homepageText is a non-false value.
-    for (key in this.config.source) {
+    if (fs.existsSync(this.config.homepage) && typeof this.config.homepage !== 'undefined') {
       if (!homepageText) {
         try {
-          files = glob.sync(this.config.source[key] + '/**/' + this.config.homepage);
+          files = glob.sync(this.config.homepage);
           if (files.length) {
             homepageText = ' ' + marked(fs.readFileSync(files[0], 'utf8'));
+            console.log('... parsing content from ' + this.config.homepage);
           }
         } catch (e) {
           // empty
+          console.log('There was an error reading the content in' + this.config.homepage);
```

Comment:
This change looks unrelated to the problem you describe.

Also, it's wrong. We need to search for the homepage in each of the source directories given since there can be multiple directories specified on the command line, e.g. `--source dir1 --source dir2`